mantidproject/mantid
# Fix OSIRIS diffspec vanadium and calibration error

### Description of work
This PR fixes an error when attempting to run an OSIRIS diffonly diffraction reduction with both a vanadium file and a calibration file selected. The error was caused by the vanadium file not being calibrated like the sample file is.

This PR also does a large simplification of the widgets used in the Diffraction interface, allowing me to remove some overly complex code.

This PR also replaces the use of deprecated `AlignDetectors` with
```py
ApplyDiffCal(...)
ConvertUnits(..., Target="dSpacing")
ApplyDiffCal(..., ClearCalibration=True)
```

Fixes #36940



#### Further detail of work


### To test:
1. Open `Indirect`->`Diffraction` interface
2. Choose the `IRIS` instrument. Check that the calibration option is not visible
3. Repeat for `TOSCA` and `VESUVIO`
4. Choose the `OSIRIS` instrument. The calibration option should be visible
5. Choose `diffspec`. The calibration and vanadium options should be unticked by default
6. Choose `diffonly`. The calibration and vanadium options should be permanently ticked (mandatory).

For OSIRIS diffonly:
1. Load Sample 137793
2. Load Vanadium 137713
3. Load Calibration file below
[osiris_041_RES10.zip](https://github.com/mantidproject/mantid/files/14572095/osiris_041_RES10.zip)

5. Click `Run`
6. The algorithm should run successfully
7. Repeat for OSIRIS "diffspec" mode

Choose IRIS:
1. Load Sample 26176
2. Click `Run`
3. The algorithm should run successfully

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
